app/assets/src/components/views/PipelineViz.jsx

```diff
@@ -25,29 +22,112 @@ class PipelineViz extends React.Component {
       ...(props.admin ? ["Experimental"] : []),
     ];
 
+    this.pipelineVersion = this.props.stageResults.pipeline_version;
+    this.stagesData = this.getStagesData();
+    this.graphs = [];
+    this.graphContainers = [];
+
     this.state = {
       stagesOpened: [true, true, true, true],
       zoom: 1,
+      sidebarVisible: false,
+      sidebarParams: {},
     };
   }
 
   componentDidMount() {
     this.drawGraphs();
   }
 
-  handleMouseWheelZoom = e => {
-    const { zoomChangeInterval } = this.props;
-    const zoomChange = (e.deltaY < 0 ? 1 : -1) * zoomChangeInterval;
-    this.setState({ zoom: this.state.zoom + zoomChange });
-  };
+  getStagesData() {
+    // TODO(ezhong): Include file download urls once passed up from backend.
+    const stageResults = this.stageResultsWithModifiedStepNames();
+    const filePathToOutputStep = this.generateFilePathToOutputStep();
+
+    const stages = Object.keys(stageResults).map(stageName => {
+      const rawStageData = stageResults[stageName];
+      const steps = rawStageData.steps.map(step => {
+        const name = step.class;
+
+        const inputInfo = step.in
+          .map(inTarget => {
+            const inTargetFiles = rawStageData.targets[inTarget].map(
+              fileName => {
+                let filePath;
+                if (inTarget in rawStageData.given_targets) {
+                  filePath = `${
+                    rawStageData.given_targets[inTarget].s3_dir
```

Comment:
you might want some handling for if s3_dir / output_dir_s3 / etc. are null or "" for whatever reason

Reply:
Should be handled in `createFilePath`